arangod/RocksDBEngine/RocksDBRecoveryManager.cpp

```diff
@@ -375,49 +385,76 @@ class WBReader final : public rocksdb::WriteBatch::Handler {
 };
 
 /// parse the WAL with the above handler parser class
-bool RocksDBRecoveryManager::parseRocksWAL() {
-  RocksDBEngine* engine =
-      static_cast<RocksDBEngine*>(EngineSelectorFeature::ENGINE);
-  for (auto helper : engine->recoveryHelpers()) {
-    helper->prepare();
-  }
+Result RocksDBRecoveryManager::parseRocksWAL() {
+  Result rv;
+  std::unique_ptr<WBReader> handler;
 
-  // Tell the WriteBatch reader the transaction markers to look for
-  auto handler =
-      std::make_unique<WBReader>(engine->settingsManager()->counterSeqs());
+  try {
+    RocksDBEngine* engine =
+        static_cast<RocksDBEngine*>(EngineSelectorFeature::ENGINE);
+    for (auto helper : engine->recoveryHelpers()) {
+      helper->prepare();
+    }
 
-  auto minTick = std::min(engine->settingsManager()->earliestSeqNeeded(),
-                          engine->releasedTick());
-  std::unique_ptr<rocksdb::TransactionLogIterator> iterator;  // reader();
-  rocksdb::Status s = _db->GetUpdatesSince(
-      minTick, &iterator, rocksdb::TransactionLogIterator::ReadOptions(true));
-  if (!s.ok()) {  // TODO do something?
-    return false;
-  }
+    // Tell the WriteBatch reader the transaction markers to look for
+    handler = std::make_unique<WBReader>(engine->settingsManager()->counterSeqs());
 
-  while (iterator->Valid()) {
-    s = iterator->status();
-    if (s.ok()) {
-      rocksdb::BatchResult batch = iterator->GetBatch();
-      handler->currentSeqNum = batch.sequence;
-      s = batch.writeBatchPtr->Iterate(handler.get());
-    }
-    if (!s.ok()) {
-      LOG_TOPIC(ERR, Logger::ENGINES) << "error during WAL scan";
-      break;
+    auto minTick = std::min(engine->settingsManager()->earliestSeqNeeded(),
+                            engine->releasedTick());
+    std::unique_ptr<rocksdb::TransactionLogIterator> iterator;  // reader();
+    rocksdb::Status s = _db->GetUpdatesSince(
+        minTick, &iterator, rocksdb::TransactionLogIterator::ReadOptions(true));
+
+    rv = rocksutils::convertStatus(s);
+
+    if(rv.ok()){
+      while (iterator->Valid()) {
+        s = iterator->status();
+        if (s.ok()) {
+          rocksdb::BatchResult batch = iterator->GetBatch();
+          handler->currentSeqNum = batch.sequence;
+          s = batch.writeBatchPtr->Iterate(handler.get());
+        }
+
+        std::string const msg = "error during WAL scan ";
+
+        if (!s.ok()) {
+          LOG_TOPIC(ERR, Logger::ENGINES) << msg;
+          rv = rocksutils::convertStatus(s);
+          rv.reset(rv.errorNumber(), msg + rv.errorMessage());
```

Comment:
why not put "error during WAL scan " + rv.errorMessage() into msg, so it can be both logged and returned as message in `rv`?
If done like above, the log will only say "error during WAL scan ". This can easily be improved by putting the error message into the log as well.